x-pack/plugins/cases/server/connectors/cases/schema.ts

```diff
@@ -42,4 +46,29 @@ export const CasesConnectorRunParamsSchema = schema.object({
   groupingBy: GroupingSchema,
   owner: schema.string(),
   rule: RuleSchema,
+  timeWindow: schema.string({
+    defaultValue: '7d',
+    validate: (value) => {
+      /**
+       * Validates the time window.
+       * Acceptable format:
+       * - First character should be a digit from 1 to 9
+       * - All next characters should be a digit from 0 to 9
+       * - The last character should be d (day) or w (week) or M (month) or Y (year)
+       *
+       * Example: 20d, 2w, 1M, etc
+       */
+      const timeWindowRegex = new RegExp(/[1-9][0-9]*[d,w,M,y]/, 'g');
```

Comment:
Should you add the start and end of string anchors? `^[1-9][0-9]*[d,w,M,y]$`

Reply:
Thank you for explaining the issue offline. Good question! I tested and I did not see any issues. I will put more tests in my next PR to cover more scenarios.

Reply:
We talked about it again offline. I was wrong. @adcoelho found an issue that the user could bypass the Regex. Thank you so much! I fixed it.